File: itmd-521/labs/week-05/week05.scala
package main.scala.chapter3
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.types._

object week05 {
    def main(args: Array[String]) {

        val spark = SparkSession
          .builder
          .appName("week05")
          .getOrCreate()
          
        if (args.length <= 0){
            println("Usage: DivvySet <Divvy_file_dataset.csv>")
            System.exit(1)
        }
    
        // Infering the Schema
        val data_source_file=args(0)
        val infer_DF = spark.read.format("csv") .option("header", "true") .option("inferSchema", "true") .load(data_source_file)
        infer_DF.show(false)
        println(infer_DF.printSchema)
        println("Infering the Schema In Scala. The number of records in this DataFrame is: "+ infer_DF.count())
        println("---------------------------------------------------------------------------------------------------------------")
        
        // Schema programmatically use StructFields
        
        val struct_schema = StructType(Array(StructField("trip_id",IntegerType,false),
        StructField("starttime",StringType,false),
        StructField("stoptime",StringType,false),
        StructField("bikeid",IntegerType,false), 
        StructField("tripduration",IntegerType,false),
        StructField("from_station_id",StringType,false),
        StructField("from_station_name",StringType,false),
        StructField("to_station_id",StringType,false),
        StructField("to_station_name",StringType,false),
        StructField("usertype",StringType,false),
        StructField("gender",StringType,false),
        StructField("birthyear",IntegerType,false)
        ))
        val structure_divvy_DF=spark.read.schema(struct_schema).format("csv").option("header","true").option("structureSchema","true").load(data_source_file)
        structure_divvy_DF.show(false)
        println(structure_divvy_DF.printSchema)
        println("Schema programmatically use StructFields. The number of records in this DataFrame is: "+ structure_divvy_DF.count())
        println("---------------------------------------------------------------------------------------------------------------")

        // Attaching a schema via DLL and reading the csv

        val schema_DDL= "trip_id INT, starttime STRING,stoptime STRING,bikeid INT,tripduration INT,from_station_id INT ,from_station_name STRING,to_station_id INT ,to_station_name STRING,usertype STRING,gender STRING,birthyear INT"
        val DDL_DF = (spark.read.schema(schema_DDL).format("csv")).option("header", "true").load(data_source_file)
        DDL_DF.show(false)
        print(DDL_DF.printSchema)
        println(DDL_DF.count())
        println("---------------------------------------------------------------------------------------------------------------")

    } 
}

File: itmd-521/labs/week-05/week05.py

import sys
from pyspark.sql import SparkSession
from pyspark.sql.types import *

if __name__ == "__main__":
    if len(sys.argv) != 2:
        sys.exit(-1)

    spark = (SparkSession.builder.appName("week05").getOrCreate())
    data_source_file = sys.argv[1]
    
    #inferring the Schema
    infer_DF= (spark.read.format("csv").option("header","true").option("inferSchema","true").load(data_source_file))
    infer_DF.show(n=15)
    infer_DF.printSchema()
    infer_DF.count()
    
    # Schema programmatically use StructFields
    structure_schema = StructType([StructField("trip_id", IntegerType()),
        StructField("starttime", StringType()),
        StructField("stoptime", StringType()),
        StructField("bikeid", IntegerType()),
        StructField("tripduration", IntegerType()),
        StructField("from_station_id", IntegerType()),
        StructField("from_station_name", StringType()),
        StructField("to_station_id", IntegerType()),
        StructField("to_station_name", StringType()),
        StructField("usertype", StringType()),
        StructField("gender", StringType()),
        StructField("birthyear", IntegerType())])
    
    struc_Divvy_DF=(spark.read.schema(structure_schema).format("csv")).option("header","true").option("structureSchema","true").load(data_source_file)
    struc_Divvy_DF.show()
    struc_Divvy_DF.printSchema()
    struc_Divvy_DF.count()
    
    #attaching a schema via DDL
    schema_DDL= "trip_id INT,starttime STRING,stoptime STRING,bikeid INT,tripduration INT,from_station_id INT,from_station_name STRING,to_station_id INT,to_station_name STRING,usertype STRING,gender STRING,birthyear INT"
    DDL_Df= (spark.read.schema(schema_DDL).format("csv")).option("header", "true").load(data_source_file)
    DDL_Df.show()
    DDL_Df.printSchema()
    DDL_Df.count()

    
    
    
    